zerver/decorator.py

```diff
@@ -155,6 +154,8 @@ def validate_api_key(role, api_key):
         raise JsonableError(reason + " for role '%s'" % (role,))
     if not profile.is_active:
         raise JsonableError("Account not active")
+    if is_webhook and not profile.is_incoming_webhook:
+        raise JsonableError("Account is not valid to post webhook messages")
```

Comment:
Shouldn't this be `if profile.is_incoming_webhook and not is_webhook` to preserve backwards-compatibility?

The goal here of these decorator checks is to make the incoming webhook bots effectively only able to send webhook messages, so that the API keys for these bots are less sensitive from a security perspective (currently, all bots have read access to basically every Zulip message).  It's not important to prevent the default bot type from sending webhook messages.